doc: added some warnings for buffer and array buffer factory method. (#929)
nodejs/node-addon-api: doc/array_buffer.md

expects it to be valid for the lifetime of the instance. Since the
`Napi::ArrayBuffer` is subject to garbage collection this overload is only
suitable for data which is static and never needs to be freed.

> This factory method will not provide the caller with an opportunity to free the 

Reviewer: ```suggestion

This factory method will not provide the caller with an opportunity to free the 
```

> data when hen the `Napi::ArrayBuffer` gets garbage-collected. If you need to

Reviewer: ```suggestion
data when the `Napi::ArrayBuffer` gets garbage-collected. If you need to
```

> data when they get garbage-collected. If you need to free the data retained by

Reviewer: ```suggestion
data when the `Napi::ArrayBuffer` gets garbage-collected. If you need to free the data retained by
```

the `Napi::ArrayBuffer` object please use other variants of the 

> `Napi::ArrayBuffer::New` factory method that accept `Napi::Finalizer` a 

Reviewer: ```suggestion
`Napi::ArrayBuffer::New` factory method that accept `Napi::Finalizer`, which is a 
```

> `Napi::Finalizer`, which is a function that will be invoked when  the

Reviewer: ```suggestion
`Napi::Finalizer`, which is a function that will be invoked when the
```

> function that will be invoked when `Napi::ArrayBuffer` object has been 

Reviewer: ```suggestion
function that will be invoked when the `Napi::ArrayBuffer` object has been 
```
